x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureopenai/AzureOpenAiSecretSettings.java

```diff
@@ -59,14 +63,24 @@ public static AzureOpenAiSecretSettings fromMap(@Nullable Map<String, Object> ma
         return new AzureOpenAiSecretSettings(secureApiToken, secureEntraId);
     }
 
-    public AzureOpenAiSecretSettings {
+    public AzureOpenAiSecretSettings(@Nullable SecureString apiKey, @Nullable SecureString entraId) {
         Objects.requireNonNullElse(apiKey, entraId);
```

Comment:
FYI - this was set up as require NonNullElse specifically to ensure only either the ApiKey or the Entra ID are supplied -- we need to ensure we cannot have both.

Reply:
Yes, this is checked [here](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureopenai/AzureOpenAiSecretSettings.java#L43-L47)